manifests/install/darwin.pp

```diff
@@ -1,26 +1,33 @@
-# == Class puppet_agent::install::darwin
-#
-# Private class called from puppet_agent class
-#
-# Manage the install process for Darwin OSes specifically
+# @summary Manage the install process for Darwin OSes specifically.
+# Private class called from puppet_agent class.
 #
+# @param package_version
+#   The puppet-agent version to install.
+# @param install_options
+#   An array of additional options to pass when installing puppet-agent. Each option in
+#   the array can either be a string or a hash. Each option will automatically be quoted
+#   when passed to the install command. With Windows packages, note that file paths in an
+#   install option must use backslashes. (Since install options are passed directly to
+#   the installation command, forward slashes won't be automatically converted like they
+#   are in `file` resources.) Note also that backslashes in double-quoted strings _must_
+#   be escaped and backslashes in single-quoted strings _can_ be escaped.
 class puppet_agent::install::darwin (
-  $package_version,
-  $install_options = [],
+  Optional $package_version,
```

Comment:
Is `package_version` actually optional? I assume it's interpolated when rendering the template:

```puppet
template('puppet_agent/do_install.sh.erb')
```